pkg/cvo/metrics.go

```diff
@@ -145,28 +146,35 @@ func shutdownHttpServer(parentCtx context.Context, svr *http.Server) {
 	}
 }
 
+const asyncResultHttpsServerName = "HTTPS server"
+
 func startListening(svr *http.Server, tlsConfig *tls.Config, lAddr string, resultChannel chan asyncResult) {
 	tcpListener, err := net.Listen("tcp", lAddr)
 	if err != nil {
-		resultChannel <- asyncResult{name: "HTTPS server", error: err}
+		resultChannel <- asyncResult{name: asyncResultHttpsServerName, error: err}
+		return
 	}
 	tlsListener := tls.NewListener(tcpListener, tlsConfig)
 	klog.Infof("Metrics port listening for HTTPS on %v", lAddr)
 	err = svr.Serve(tlsListener)
-	resultChannel <- asyncResult{name: "HTTPS server", error: err}
+	resultChannel <- asyncResult{name: asyncResultHttpsServerName, error: err}
 }
 
-func handleServerResult(result asyncResult, lastLoopError error) error {
+func handleServerResult(result asyncResult, lastLoopError error) (error, bool) {
 	lastError := lastLoopError
-
+	addressAlreadyInUse := isAddressAlreadyInUse(result)
 	// API states "Serve always returns non-nil error" but check in case API changes.
 	if result.error == nil || result.error == http.ErrServerClosed {
 		klog.Infof("Collected metrics %s goroutine.", result.name)
 	} else {
-		klog.Errorf("Collected metrics %s goroutine: %v", result.name, result.error)
+		if addressAlreadyInUse {
+			klog.Errorf("A network address reserved for cluster-version-operator to collect metrics is not available: %v", result.error)
```

Comment:
this bit seems a bit fiddly; do we need to special-case the handling of this error here?  Or can we wrap the underlying library error with some context that the existing, generic `Collected metrics %s goroutine: %v` wrapping here would still make the failure mode sufficiently clear to anyone dropping down as far as logs?

Reply:
wrapped the underlying lib error.